src/chapters/unit-testing/exercises.rst

```diff
@@ -0,0 +1,326 @@
+
+Manual Testing
+---------------
+
+Assume that a team member wrote the following code:
+
+.. replit:: js
+   :slug: UnitTestingExercises01
+   :linenos:
+
+   function createLaunchOutput(entry) {
+      let output = "";
+
+      entry = Number(entry);
+
+      if (entry%2 === 0){
+         output = "Crew ready for launch. Shuttle offline.";
+      } else if (entry%3 === 0){
+         output = "Shuttle ready for launch. Crew asleep.";
+      } else if (entry%2 === 0 && entry%3 === 0){
+         output = "Crew and shuttle ready. Cleared for launch!";
+      }
+
+      return output;
+   }
+
+   const input = require(`readline-sync`);
+
+   let response = input.question("Enter the launch code (must be an integer): ");
+
+   console.log(createLaunchOutput(response));
+
+The program must accomplish the following:
+
+#. If the user enters a number divisible only by 2, print ``'Crew ready for
+   launch. Shuttle offline.'``
+#. If the user enters a number divisible only by 3, print ``'Shuttle ready for
+   launch. Crew asleep.'``
+#. If the user enters a number divisible by both 2 and 3, print ``'Crew and
+   shuttle ready. Cleared for launch!'``
+#. If the user enters a number that is NOT divisible by 2 OR 3, print ``'Crew
+   and shuttle unresponsive. Launch aborted!'``
+#. If the user enters a string, a decimal, or just hits return, print ``'ARRR!
+   Raid yonder shuttle!'``
+
+Test for Errors
+^^^^^^^^^^^^^^^^
+
+#. ``createLaunchOutput`` runs, so there are no syntax errors, but does the
+   program produce the correct output?
+
+   a. Consider the following inputs: 22, 33, 6, 0, and 7. What output should
+      the program produce in each case?
+   b. Run the code as-is and *test* it with each of the inputs. Do any of the
+      values produce incorrect output?
+
+#. Using the techniques you already learned, fix the code so that any integer
+   input produces the correct output.
+
+#. We *tested* the code with integers. What if the user types in something
+   else?
+
+   a. Add the following validation check to the ``createLaunchOutput``
+      function:
+
+      .. sourcecode:: js
+         :linenos:
+
+         function createLaunchOutput(entry) {
+            let output = "";
+
+            if (!entry.includes('.') || isNaN(entry) || entry === ' '){
+               output = "ARRR! Raid yonder shuttle!";
+               return output;
+            }
+
+            entry = Number(entry);
+         etc...
+
+   b. Run the updated program and *test* it with the inputs: 2.22, 'H', and ''.
+      Which entries produce incorrect output?
+   c. Fix the new code so that it correctly returns ``'ARRR! Raid yonder
+      shuttle!'`` for the empty string, decimal values, and non-numerical
+      inputs.
+
+#. Once the code returns the correct output for all of the suggested input
+   values, continue to *test* it with your own entries. Make sure your fixes
+   all work.
+
+Good! You are now ready to *automate* the testing process.
+
+Automatic Testing
```

Comment:
This is also going to hurt. In my opinion this section is also too long and more like a book section than an exercise. I would remove this as well. More about this in my overall comment.

Reply:
I concur. This replicates the examples from the book.  We can easily remove this part.